src/WebJobs.Extensions.DurableTask/Listener/TaskOrchestrationShim.cs

```diff
@@ -165,4 +225,36 @@ public override void RaiseEvent(OrchestrationContext unused, string eventName, s
             this.context.RaiseEvent(eventName, serializedEventData);
         }
     }
+
+    internal class AsyncCommand
```

Comment:
Does this (and `AsyncActionType`) need to be an `internal` class? Could we instead make it a private inner class? The C# purist in me who studied the old books on .NET Framework guidelines wants to put internal classes into their own files. But if these classes are only used here, then it makes sense to just make them private inner classes.